src/plugins/intel_gna/src/log/log.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ostream>
+#include <iostream>
+
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_gna {
+
+/**
+ * @brief Log manager. It supports all ov::log::Level values.
+ * To use it need to set log level then just call one of the logging methods:
+ *      GnaLog::Gnalog(ov::log::Level::DEBUG);
+ *      GnaLog::info()  << "log message"; // this message will be printed
+ *      GnaLog::trace() << "log message"; // this message shoudl not be printed 
+ */
+class GnaLog {
+ private :
+    GnaLog() {}
+
+    static GnaLog& log(ov::log::Level log_level) {
+        GnaLog& obj = get_instance();
+        obj.message_level_ = log_level;
+        obj << "[" << log_level << "]" << " ";
+        return obj;
+    }
+
+    /** Configuration log level. It should be set in the plugin configuration */
+    ov::log::Level log_level_ = ov::log::Level::NO;
+
+    /** Log level of particular log message */
+    ov::log::Level message_level_ = ov::log::Level::NO;
+
+ public :
+    GnaLog(GnaLog const&) = delete;
+    void operator = (const GnaLog&) = delete;
+
+    GnaLog(ov::log::Level log_level) {
+        get_instance().log_level_ = log_level;
+    }
+
+    static GnaLog& get_instance() {
+        static GnaLog log_obj;
+        return log_obj;
+    }
+
+    /**
+     * @brief Set ERROR log level
+     * @return GnaLog object
+     */
+    static GnaLog& error() {
+        return log(ov::log::Level::ERR);
+    }
+
+    /**
+     * @brief Set WARNING log level
+     * @return GnaLog object
+     */
+    static GnaLog& warning() {
+        return log(ov::log::Level::WARNING);
+    }
+
+    /**
+     * @brief Set DEBUG log level
+     * @return GnaLog object
+     */
+    static GnaLog& debug() {
+        return log(ov::log::Level::DEBUG);
+    }
+
+    /**
+     * @brief Set INFO log level
+     * @return GnaLog object
+     */
+    static GnaLog& info() {
+        return log(ov::log::Level::INFO);
+    }
+
+    /**
+     * @brief Set TRACE log level
+     * @return GnaLog object
+     */
+    static GnaLog& trace() {
+        return log(ov::log::Level::TRACE);
+    }
+
+    template <class T>
+    GnaLog &operator << (const T &obj) {
+        if (message_level_ <= log_level_) {
+            if (message_level_ == ov::log::Level::ERR) {
+                std::cerr << obj;
+            } else {
+                std::cout << obj;
```

Comment:
IMO we should use `cout` for all,
otherwise, it will be harder to synchronize if not impossible.
optionally I would consider to duplicate ERR (write it to both streams).

Reply:
No we should not duplicate error stream, because this is normal for user to get the error stream independently and forward it to out if needed.
Not sure about writing error messages to std::out instead of std::error.